archive_repo.py

```diff
@@ -0,0 +1,667 @@
+
+    issue_cursor = None
+    get_more_issues = True
+
+    # Since PRs can't be filtered by their update time, we retrieve
+    # them in update-time order and cut off pagination once we're
+    # older than the reference file.
+
+    while get_more_issues:
+        query = gql_AllPRs_Initial
+        variables = {"owner": owner, "repo": repo}
+        if issue_cursor is not None:
+            query = gql_AllPRs_Subsequent
+            variables["cursor"] = issue_cursor
+
+        data = submit_query(query, variables, "pull requests")
+
+        # Iterate through the PRs
+        prs = data["repository"]["pullRequests"]
+
+        for pr in prs["nodes"]:
+            number = pr["number"]
+
+            # Since we can't filter, check if we already have this one.
+            if number in pr_ref:
+                ref_updatedAt = dp.parse(pr_ref[number]["updatedAt"])
+                dl_updatedAt = dp.parse(pr["updatedAt"])
+                if ref_updatedAt >= dl_updatedAt:
+                    continue
+
+            # Issues only have comments; PRs have both comments and reviews,
+            # and reviews themselves have comments.
+            followPagination(
+                pr,
+                "comments",
+                gql_PR_Comments_Query,
+                f"additional comments on PR#{number}",
+            )
+            followPagination(
+                pr, "reviews", gql_PR_Review_Query, f"additional reviews on PR#{number}"
+            )
+
+            for review in pr["reviews"]["nodes"]:
+                followPagination(
+                    review,
+                    "comments",
+                    gql_PR_ReviewComments_Query,
+                    f"additional review comments on PR#{number}",
+                )
+
+            # Delete the old instance; add this instance
+            if number in pr_ref:
+                del pr_ref[number]
+            pr_ref[number] = pr
+            just_copy_old_file = False
+
+        get_more_issues = prs["pageInfo"]["hasNextPage"]
+        issue_cursor = prs["pageInfo"]["endCursor"]
+
+        # Stop paginating if we've caught up to the last download
+        if prs["nodes"] and lastSuccess and not ref_is_issues_only:
+            oldestRetrieved = dp.parse(prs["nodes"][-1]["updatedAt"])
+            if oldestRetrieved < lastSuccess:
+                get_more_issues = False
+
+## Ready to output
+
+## Pick up everything in the reference if nothing new was downloaded
+if just_copy_old_file:
+    shutil.copyfile(args.refFile, args.outFile)
+else:
+    output = {
+        "magic": "B8n2c@e8kvfx",
+        "timestamp": now.isoformat(),
+        "version": "1.0",
+        "repo": args.repo,
+        "issues": [issue for (id, issue) in sorted(issue_ref.items())],
+    }
+    if not args.issuesOnly:
+        output["pulls"] = [pr for (id, pr) in sorted(pr_ref.items())]
+
+    with open(args.outFile, "w") as output_file:
+        json.dump(output, output_file, indent=4)
```

Comment:
```suggestion
        json.dump(output, output_file, indent=2)
```

Less side-scrolling :)